.github/workflows/build.yml

```diff
@@ -1,111 +1,94 @@
 
-    - name: Install GitVersion
-      uses: gittools/actions/gitversion/setup@v0.9.7
-      with:
-        versionSpec: '5.x'
-
-    - name: Determine Version
-      id:   gitversion
-      uses: gittools/actions/gitversion/execute@v0.9.7
-      with:
-        useConfigFile: true
-
-    - name: Set version variables
-      run: |
-        echo "STABLE=${{steps.gitversion.outputs.majorMinorPatch}}" >> $GITHUB_ENV
-        echo "BETA=${{steps.gitversion.outputs.majorMinorPatch}}-beta.$(date +%s)" >> $GITHUB_ENV
+  Test:
+    runs-on: ubuntu-latest
+    if: inputs.project-to-test != ''
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
 
-    - name: Output Version Numbers
-      id: version
-      run: |
-        echo "::set-output name=beta-version::${{env.BETA}}"
-        echo "::set-output name=stable-version::${{env.STABLE}}"
+      - name: Setup .NET
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 6.0.x
 
-    - name: Pack
-      run: |
-        dotnet pack "${{inputs.project-to-pack}}" --configuration Release /p:Version="${{env.BETA}}" --output unsigned/beta
-        dotnet pack "${{inputs.project-to-pack}}" --configuration Release /p:Version="${{env.STABLE}}" --output unsigned/stable
+      - name: Test
+        run: dotnet test "${{inputs.project-to-test}}" --filter "${{inputs.test-filter}}" /p:CollectCoverage=true /p:Threshold=${{inputs.coverage-threshold}} /p:ThresholdType=line /p:ThresholdStat=total /p:CoverletOutputFormat=cobertura /p:CoverletOutput=$GITHUB_WORKSPACE/coverage.xml
 
-    - name: Upload Unsigned Packages
-      uses: actions/upload-artifact@v2
-      with:
-        name: unsigned
-        path: unsigned
+      - name: Generate Test Coverage Report
+        uses: danielpalme/ReportGenerator-GitHub-Action@5.1.3
+        with:
+          reports: "coverage.xml"
+          targetdir: "coveragereport"
+          reporttypes: "HtmlInline;Cobertura;Badges" # The output formats and scope (separated by semicolon) Values: Badges, Clover, Cobertura, CsvSummary, Html, Html_Dark, Html_Light, HtmlChart, HtmlInline, HtmlInline_AzurePipelines, HtmlInline_AzurePipelines_Dark, HtmlInline_AzurePipelines_Light, HtmlSummary, JsonSummary, Latex, LatexSummary, lcov, MarkdownSummary, MHtml, PngChart, SonarQube, TeamCitySummary, TextSummary, Xml, XmlSummary
+          verbosity: "Info"
+          tag: "${{ github.run_number }}_${{ github.run_id }}"
+          toolpath: "reportgeneratortool"
+        if: always()
 
-#    - name: Get branch names
-#      id: branch-names
-#      uses: tj-actions/branch-names@v5.1
+      - name: Upload Coverage Report Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: CoverageReport
+          path: coveragereport
+        if: always()
 
-#    - name: Current branch name
-#      run: echo "${{steps.branch-names.outputs.current_branch}}"
+  Verify:
```

Comment:
I don't feel that including this job in the build workflow makes sense. When you split the build-and-pack flow into build and pack flows, you're implying that they could be used independently. So Build could be used with non-packable projects in which case this package verification job doesn't make sense.
I feel that verification is part of the pack-ing process, and if you want to run the verification task in the PR build runs, then maybe PRs should run both Build and Pack workflows for packable projects - and I see no harm in that since all Pack is doing is pack-ing and uploading the unsigned artifact, which is arguably something you want in PRs anyways to enable developers to download the package and test it with other solutions.

Reply:
So if you move this job to the Pack workflow, you don't even need to pack twice, you can just download the artifact and verify it

Reply:
So, I am quite firm about the choice to move the Verify piece during PR time. I'm fine with making it only happen if the project has the <IsPackable> metadata in the .csproj or making it configurable in some way though. I already had to experience the annoyance of making a new PR after discovering the verify step was complaining I had missing metadata, so why wait until after the PR to find that out?

Reply:
Made this optional/configurable

Reply:
Sorry, let me clarify: I'm not saying we shouldn't verify during PR time - I'm saying Pack and Verify should be one thing and whether you run Build and Pack during PR or run Build only is another story. 
These re-usable pieces do not enforce when they're called, and we should treat them as such. The name (Build) doesn't imply any form of the project to be triggered for, and that's why you chose not to define `project-to-pack` input for it. So the word "pack" shouldn't appear in this workflow. Making it configurable also doesn't feel right, because it doesn't belong here.
Pack workflow on the other hand is designed specifically for those kinds of projects and for that specific functionality. Packing without verifying is not enough since we know that there are certain metadata rules that need to be verified.

If we move the Verify piece to Pack workflow we can still run Pack during PRs and get two benefits:
1. verify package metadata.
2. get an unofficial NuGet file to use for testing if needed.